temporal-sdk/src/main/java/io/temporal/internal/worker/StickyQueueBalancer.java

```diff
@@ -43,35 +40,35 @@ public StickyQueueBalancer(int pollersCount, boolean stickyQueueEnabled) {
   /**
    * @return task queue kind that should be used for the next poll
    */
-  public TaskQueueKind makePoll() {
+  public synchronized TaskQueueKind makePoll() {
```

Comment:
Usually linters will tell you not to add this keyword but instead make a private final object that you wrap in `synchronized (theObject) {` because it can be unclear to users of this class that they can cause deadlocks if they try to also synchronize on this instance. Probably fine in internal code though.